functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp

```diff
@@ -176,6 +182,181 @@ householder_product_batch_rule(const Tensor &input, c10::optional<int64_t> input
+      T... extra_args) {
+    TORCH_CHECK(!first_bdim.has_value() || first.dim() > 2,
+              op_name, ": The input tensor A must have at least 2 dimensions.");
+    TORCH_CHECK(!second_bdim.has_value() || second.dim() > 2,
+              op_name, ": The input tensor A must have at least 2 dimensions.");
+    const auto tensor_other = _binary_pointwise_helper(first, first_bdim, second, second_bdim);
+    const auto tensor_ = std::get<0>(tensor_other);
+    const auto other_ = std::get<1>(tensor_other);
+    const auto res = Func(tensor_, other_, std::forward<T>(extra_args)...);
+    return std::make_tuple(std::get<0>(res), 0, std::get<1>(res), 0);
+  }
+};
+
+oneOutput cholesky_solve_batch_rule(
+    const Tensor& self, c10::optional<int64_t> self_bdim,
+    const Tensor& A, c10::optional<int64_t> A_bdim,
+    bool upper) {
+  TORCH_CHECK(!self_bdim.has_value() || self.dim() > 2,
+           "b should have at least 2 dimensions, but has ", self.dim(), " dimensions instead");
+  TORCH_CHECK(!A_bdim.has_value() || A.dim() > 2,
+           "u should have at least 2 dimensions, but has ", A.dim(), " dimensions instead");
+
+  const auto tensor_other = _binary_pointwise_helper(self, self_bdim, A, A_bdim);
+  const auto tensor_ = std::get<0>(tensor_other);
+  const auto other_ = std::get<1>(tensor_other);
+  return std::make_tuple(at::cholesky_solve(tensor_, other_, upper), 0);
+}
+
+threeOutputs linalg_lu_factor_ex_batch_rule(
+    const Tensor& A, c10::optional<int64_t> A_bdim, bool pivot, bool check_errors) {
+  TORCH_CHECK(A.dim() > 2, "torch.lu_factor: Expected tensor with 2 or more dimensions. Got size: ", A.sizes(), " instead");
+  const auto A_ = moveBatchDimToFront(A, A_bdim);
+  const auto res = at::linalg_lu_factor_ex(A_, pivot, check_errors);
+  return std::make_tuple(std::get<0>(res), 0, std::get<1>(res), 0, std::get<2>(res), 0);
+}
+
+#define LINALG_CHECK_MATRIX_UNARY_BATCH_RULE(fn, num_out) SINGLE_ARG(\
+  LinalgCheckMatrixUnaryRuleHelper<\
+    func_string_##fn,\
+    decltype(&ATEN_FN(fn)),\
+    &ATEN_FN(fn),\
+    c10::guts::function_traits<decltype(ATEN_FN(fn))>::parameter_types>::apply_##num_out)
+
+#define LINALG_CHECK_MATRIX_UNARY_BATCH_RULE2(fn, overload, num_out) SINGLE_ARG(\
+  LinalgCheckMatrixUnaryRuleHelper<\
+    func_string_##fn_##overload,\
+    decltype(&ATEN_FN2(fn, overload)),\
+    &ATEN_FN2(fn, overload),\
+    c10::guts::function_traits<decltype(ATEN_FN2(fn, overload))>::parameter_types>::apply_##num_out)
+
+#define LINALG_CHECK_MATRIX_BINARY_BATCH_RULE(fn, num_out) SINGLE_ARG(\
+  LinalgCheckMatrixBinaryRuleHelper<\
+    func_string_##fn,\
+    decltype(&ATEN_FN(fn)),\
+    &ATEN_FN(fn),\
+    c10::guts::function_traits<decltype(ATEN_FN(fn))>::parameter_types>::apply_##num_out)
+
+
+#define LINALG_STRING_CONST(fn, op_name) \
+  const char func_string_##fn[] = #op_name;\
+
+#define LINALG_STRING_CONST2(fn, overload, op_name) \
+  const char func_string_##fn_##overload[] = #op_name;\
```

Comment:
nit: took me a couple of minutes to figure out what we were doing with this. I would have probably just passed an extra argument to the e.g. `LINALG_CHECK_MATRIX_UNARY_ONE_OUT` macro that is the name of the operator in string form

Reply:
Okay I think this is saying one of two things so I'll try to clarify both haha. Also if this isn't what you were asking, lmk
So right now, this does
(1) declare a string constant `const char <name>[] = <function name as appear in error>`
(2) pass that constant to the template so it looks like `LinalgCheckMatrixUnaryRuleHelper<name, aten func info>`

The questions might be either:
(A) If this is saying pass the string directly as the template param instead of doing this weird declare a constant and then pass that, we can't do that (https://stackoverflow.com/questions/1826464/c-style-strings-as-template-arguments). I'll add a comment about this above these since they're confusing

(B) If this is saying instead of `LINALG_CHECK_MATRIX_UNARY_ONE_OUT(cholesky_inverse, cholesky_inverse)` we should do `LINALG_CHECK_MATRIX_UNARY_ONE_OUT(cholesky_inverse, "cholesky_inverse")` if I did this, the error would be `"cholesky_inverse": The input tensor A...`

Reply:
> A) If this is saying pass the string directly as the template param instead of doing this weird declare a constant and then pass that, we can't do that (https://stackoverflow.com/questions/1826464/c-style-strings-as-template-arguments). I'll add a comment about this above these since they're confusing

Thanks for the clarification, that's what I was asking about. I didn't know that string literals can't get passed to templates, but that makes sense.